cli/install/install.go

```diff
@@ -848,16 +914,36 @@ func installTarantool(binDir string, incDir string, installCtx InstallCtx,
 
 	// Check that the version exists.
 	if tarVersion != "master" {
-		versionFound := false
-		for _, ver := range versions {
-			if tarVersion == ver.Str {
-				versionFound = true
-				break
+		if isVersion(tarVersion) {
+			log.Infof("Searching in versions...")
+			for _, ver := range versions {
+				if tarVersion == ver.Str {
+					versionFound = true
+					break
+				}
 			}
-		}
+		} else {
+			log.Infof("Searching in commits...")
+
+			if len(tarVersion) < minCommitHashLength {
+				return fmt.Errorf("the hash must contain at least 7 characters")
+			}
+
+			commits, err := getCommits(installCtx.Local, distfiles, "tarantool")
 
-		if !versionFound {
-			return fmt.Errorf("%s version of tarantool doesn't exist", tarVersion)
+			if err != nil {
+				return err
+			}
+
+			croppedHash := tarVersion[0:minCommitHashLength]
```

Comment:
There is still a code duplication for the logic that:
- Checks version length against min commit hash len.
- Getting commits.
- Cropping the hash.
- Search cropped hash in hashes list.

Why isn't there a function to do this to avoid code duplication?

Reply:
fixed